### PR TITLE
fix(forms): support friendlier disabled binding in ngModel

### PR DESCRIPTION
**Pull request by kara**
Fixes unbound disabled attributes and 'false' with ngModel.

### CODE REVIEW

File: modules/@angular/forms/test/directives_spec.ts [hunk 1]
```diff
@@ -539,6 +539,45 @@ export function main() {
 
            expect(ngModel.control.errors).toEqual({'async': true});
          }));
+
+      it('should mark as disabled properly', fakeAsync(() => {
+           ngModel.ngOnChanges({isDisabled: new SimpleChange('', undefined)});
+           tick();
+           expect(ngModel.control.disabled).toEqual(false);
```
Comment (vicb) [inline modules/@angular/forms/test/directives_spec.ts]: replace the 3 lines with `expectControlDisabled('', undefined, false);` ?

File: modules/@angular/forms/test/template_integration_spec.ts [hunk 2]
```diff
@@ -420,6 +420,31 @@ export function main() {
...
+               template: `
+            <form>
+             <input name="name" [(ngModel)]="name" disabled>
+            </form>
+          `,
+             }
+           });
+           const fixture = TestBed.createComponent(NgModelForm);
+           fixture.detectChanges();
+           tick();
```
Comment (vicb) [inline modules/@angular/forms/test/template_integration_spec.ts]: do you need to both detectChanges & tick ?

Comment (kara) [inline modules/@angular/forms/test/template_integration_spec.ts]: Yep, both necessary

File: modules/@angular/forms/test/directives_spec.ts (hunk 1 above)
Comment (kara) [inline modules/@angular/forms/test/directives_spec.ts]: Hmm, I can see the savings in terms of DRY-ness, but from the suggested function call it's not immediately obvious what you're testing (and what the relationship is between the three values). My inclination is that it's more readable as is.

Comment (vicb) [inline modules/@angular/forms/test/directives_spec.ts]: You can find a better test description / find a better function name / add a comment. But I agree in the end it's personal pref.

File: modules/@angular/forms/test/template_integration_spec.ts (hunk 2 above)
Comment (vicb) [inline modules/@angular/forms/test/template_integration_spec.ts]: What about switching to `async()` and remove all the `tick()` then

File: modules/@angular/forms/test/template_integration_spec.ts [hunk 3]
```diff
@@ -420,6 +420,31 @@ export function main() {
            });
          }));
 
+      it('should disable a control with unbound disabled attr', fakeAsync(() => {
+           TestBed.overrideComponent(NgModelForm, {
```
Comment (vicb) [inline modules/@angular/forms/test/template_integration_spec.ts]: you can write

```ts
           const fixture = TestBed.overrideComponent(NgModelForm, {
             set: {
               template: `
            <form>
             <input name="name" [(ngModel)]="name" disabled>
           </form>`,
             }
           }).createComponent(NgModelForm);
```

File: modules/@angular/forms/test/template_integration_spec.ts (hunk 2 above)
Comment (kara) [inline modules/@angular/forms/test/template_integration_spec.ts]: I actually just switched from async to fakeAsync on @vsavkin's suggestion :) I think fakeAsync is cleaner. async would require nested whenStable() calls.

**Pull request merged**